pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -364,6 +380,30 @@ func (a Adapter) execDevfile(commandsMap common.PushCommandsMap, componentExists
 
 	}
 
+	// Get Debug Command
+	command, ok = commandsMap[versionsCommon.DebugCommandGroupType]
+	if ok && isDebug {
+		klog.V(4).Infof("Executing devfile command %v", command.Exec.Id)
+		compInfo.ContainerName = command.Exec.Component
+
+		// Check if the devfile debug component containers have supervisord as the entrypoint.
+		// Start the supervisord if the odo component does not exist
+		if !componentExists {
+			err = a.InitRunContainerSupervisord(command.Exec.Component, podName, containers)
+			if err != nil {
+				return
+			}
+		}
+
+		if componentExists && !common.IsRestartRequired(command) {
+			klog.V(4).Infof("restart:false, Not restarting debugRun Command")
+			err = exec.ExecuteDevfileRunActionWithoutRestart(&a.Client, *command.Exec, command.Exec.Id, compInfo, show)
+			return
```

Comment:
I am confused here  `ExecuteDevfileRunActionWithoutRestart`, it would start the run command if not running, and is there any use case for `restart: false` for debug commands?

Reply:
Fixed. Regarding `restart:false` for debug, I am not sure. Let's keep it as it's an option in the devfile.